keps/sig-apps/2232-jobs-stopped/README.md

```diff
@@ -0,0 +1,558 @@
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Scalability](#scalability)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed](#infrastructure-needed)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+tl;dr: make it possible to create Jobs without immediately creating Pods.
+
+The Jobs API allows users to create Pods using the jobs controller with
+specific requirements such as Pod-level parallelism, completing a certain
+number of executions, and restart policies. This KEP proposes an enhancement
+that will allow higher-level orchestrators to create Jobs with the option to
+NOT immediately create Pods until a bit is flipped. Such a Job is called a
+*Stopped Job* and is facilitated through the `stopped` field in the Job spec.
+
+## Motivation
+
+The Job controller tracks and manages all Jobs within Kubernetes. When a new
+Job is created by an user or a higher-level orchestrator and is seen by the
+controller, it will immediately begin creating pods to satisfy the requirements
+of the Job.
+
+This is sometimes undesirable if a higher-level orchestrator wants to create a
+Job in a "stopped" state. For instance, if a Job is created and there aren't
+enough resources to schedule all Pods to nodes, the Job will be in a state
+where its Pod-level parallelism requirements are not met. If the cluster is
+able to scale up its resources, the Job will eventually meet such requirements,
+but this isn't useful in cases such as all-or-nothing scheduling where Pods
```

Comment:
other than all-or-nothing, some quota-based queuing would also be possible with the stopped flag.